lib/elixir/lib/string_io.ex

```diff
@@ -263,35 +255,24 @@ defmodule StringIO do
 
   ## get_line
 
-  defp get_line(encoding, prompt,
-                %{input: input, output: output, capture_prompt: capture_prompt} = s) do
-    case :unicode.characters_to_list(input, encoding) do
-      {:error, _, _} ->
-        {{:error, :collect_line}, s}
-      {:incomplete, _, _} ->
-        {{:error, :collect_line}, s}
-      chars ->
-        {result, input} = do_get_line(chars, encoding)
-
-        s =
-          if capture_prompt do
-            %{s | output: <<output::binary, IO.chardata_to_string(prompt)::binary>>}
-          else
-            s
-          end
-
-        {result, %{s | input: input}}
+  defp get_line(encoding, prompt, %{input: input} = s) do
+    case bytes_until_eol(input, encoding, 0) do
+      {:ok, count, replace_cr} ->
+        {result, remainder} = case count do
```

Comment:
Done in d960c58bf8fda3ef2fc7285e1be9d32ee3b7d647. I've played around with an alternative implementation ([see Gist](https://gist.github.com/balexand/eada286ce46019cfac333947ed877065)) that avoids some repetition but adds a nested case.